ios/Views/PKPaymentButtonView.m

```diff
@@ -64,12 +64,16 @@ - (void)setButtonType:(NSString *) buttonType andStyle:(NSString *) buttonStyle
   PKPaymentButtonType type;
   PKPaymentButtonStyle style;
   
-  if ([buttonType isEqualToString: @"setUp"]) {
+  if ([buttonType isEqualToString: @"buy"]) {
+    type = PKPaymentButtonTypeBuy;
+  } else if ([buttonType isEqualToString: @"setUp"]) {
     type = PKPaymentButtonTypeSetUp;
-  } else if ([buttonType isEqualToString: @"continue"]) {
-    type = PKPaymentButtonTypeContinue;
   } else if ([buttonType isEqualToString: @"inStore"]) {
     type = PKPaymentButtonTypeInStore;
+  } else if ([buttonType isEqualToString: @"donate"]) {
+    type = PKPaymentButtonTypeDonate;
+  } else if ([buttonType isEqualToString: @"continue"]) {
+    type = PKPaymentButtonTypeContinue;
   } else {
     type = PKPaymentButtonTypePlain;
   }
```

Comment:
I feel like having a silent default is not the best behavior, and would lead to other people running into exactly what you dealt with.

Reply:
If by what I dealt with you're referring to the button always showing up as the plain button type if the width is too small, removing this default actually won't fix that. That behavior is defined in the native component itself and will happen regardless of what `buttonType` value we pass it.

This default just ensures that a valid button type is passed to the native component regardless of what we set the required `type` prop to here to prevent an error in the native layer.